### ISSUE TITLE
INFO command is empty

### ISSUE THREAD

**Issue by uspen** [pgbackrest/pgbackrest#1199]
Please provide the following information when submitting an issue (feature requests or general comments can skip this):

1. pgBackRest version: 2.30-1.rhel7

2. PostgreSQL version: 9.6

3. Operating system/version - if you have more than one server (for example, a database server, a repository host server, one or more standbys), please specify each: CentOS 7

4. Did you install pgBackRest from source or from a package? rpm from pgdg-common

5. Please attach the following as applicable:
    - `pgbackrest.conf` file(s)
    - `postgresql.conf` settings applicable to pgBackRest (`archive_command`, `archive_mode`, `listen_addresses`, `max_wal_senders`, `wal_level`, `port`)
    - errors in the postgresql log file before or during the time you experienced the issue
    - log file in `/var/log/pgbackrest` for the commands run (e.g. `/var/log/pgbackrest/mystanza_backup.log`)

7. Describe the issue:

If pgbackrest backup dir is empty, INFO command empty too

$ pgbackrest --config-path=/etc/pgbackrest --stanza=fake --output=json info
[]

$ touch /data/pgbackrest/backup/foo

$ pgbackrest --config-path=/etc/pgbackrest --stanza=fake --output=json info
[{"backup":[],"db":[],"name":"fake","status":{"code":1,"lock":{"backup":{"held":false}},"message":"missing stanza path"}}]
[pgbackrest-info.txt](https://github.com/pgbackrest/pgbackrest/files/5395323/pgbackrest-info.txt)

**Comment by pgstef** [comment]
Hi,

This seems normal. Have you run `pgbackrest --stanza=fake stanza-create` ? This will initiate the repo, manifests, sub-directories,...

**Comment by uspen** [comment]
no, if stanza not exist, INFO must be:
[{"backup":[],"db":[],"name":"fake","status":{"code":1,"lock":{"backup":{"held":false}},"message":"missing stanza path"}}]

**Comment by uspen** [comment]
But if I make other stanza or any file, fake stanza has a good answer (missing stanza path)

**Comment by pgstef** [comment]
Blank json output is actually `No stanzas exist in the repository.`: https://github.com/pgbackrest/pgbackrest/blob/master/src/command/info/info.c#L850.

```
$ pgbackrest info --stanza=my_stanza
No stanzas exist in the repository.
$ pgbackrest info --stanza=my_stanza --output=json
[]
```

The `info` command tries to list the stanza existing in the repo by listing the content of the backup directory: https://github.com/pgbackrest/pgbackrest/blob/master/src/command/info/info.c#L766.

If repo is empty, then `No stanzas exist in the repository.` is the expected feedback.

Adding a backup directory with some content will "create" the **foo** stanza and since you're looking for **fake**, the info command will hit this line I guess: https://github.com/pgbackrest/pgbackrest/blob/master/src/command/info/info.c#L485. Saying "there's something in the repo, but not the stanza you're looking for".

In conclusion:
* if the stanza doesn't exist --> missing stanza path
* if repo empty (no stanza) --> blank json / `No stanzas exist in the repository`

Kind regards,

**Comment by dwsteele** [comment]
It's possible that we could give a more detailed error for info when there are no stanzas but a stanza is specified. However, this would change the long-standing behavior of the info command so we are not inclined to do so.

I could definitely see this as an improvement for the next version of the info format, when we do that.

**Comment by dwsteele** [comment]
@pgstef created a PR for this (#1208), which looks reasonable, so I'm closing the issue. @uspen Please have a look at the PR.
